addons/repair/models/sale_order.py

```diff
@@ -70,7 +70,7 @@ def write(self, vals_list):
         old_product_uom_qty = {line.id: line.product_uom_qty for line in self}
         res = super().write(vals_list)
         for line in self:
-            if line.state in ('sale', 'done'):
+            if line.state in ('sale', 'done') and line.product_uom:
```

Comment:
Simply do not enter this logic if we don't write on `product_uom_qty` (not in `vals_list`, to rename to `vals`).